omod/src/main/java/org/openmrs/module/ugandaemr/web/resource/QueueStatisticResource.java

```diff
@@ -0,0 +1,197 @@
+        throw new ResourceDoesNotSupportOperationException("Operation not supported");
+    }
+
+    @Override
+    public PatientQueueStatistic getByUniqueId(String uniqueId) {
+        throw new ResourceDoesNotSupportOperationException("Operation not supported");
+    }
+
+    @Override
+    public NeedsPaging<PatientQueueStatistic> doGetAll(RequestContext context) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("Operation not supported");
+    }
+
+    @Override
+    public List<Representation> getAvailableRepresentations() {
+        return Arrays.asList(Representation.DEFAULT, Representation.FULL);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        if (rep instanceof DefaultRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("locationTag", Representation.REF);
+            description.addProperty("pending");
+            description.addProperty("serving");
+            description.addProperty("completed");
+            description.addSelfLink();
+            return description;
+        } else if (rep instanceof FullRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("locationTag", Representation.FULL);
+            description.addSelfLink();
+            description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            return description;
+        } else if (rep instanceof RefRepresentation) {
+            DelegatingResourceDescription description = new DelegatingResourceDescription();
+            description.addProperty("locationTag");
+            description.addProperty("pending");
+            description.addProperty("serving");
+            description.addProperty("completed");
+            description.addSelfLink();
+            return description;
+        }
+        return null;
+    }
+
+    @Override
+    protected void delete(PatientQueueStatistic syncFhirCase, String s, RequestContext requestContext) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("Operation not supported");
+    }
+
+    @Override
+    public void purge(PatientQueueStatistic syncFhirCase, RequestContext requestContext) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException("Operation not supported");
+    }
+
+
+    @Override
+    protected PageableResult doSearch(RequestContext context) {
+        PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+        String locationParam = context.getParameter("parent");
+        Date before = null;
+        Date after = null;
+        if (context.getParameter("before") != null) {
+            before = getDateFromString(context.getParameter("before"), "yyyy-MM-dd");
+        }
+
```

Comment:
@slubwama  is this a typo in the if condition: "context.getParameter("before") != null" this is checked twice instead of checking for after.